doc/harmony-ap_logging_user_guide.md

```diff
@@ -0,0 +1,371 @@
+- **Kibana**: A visualization tool for exploring and analyzing log data stored in Elasticsearch.
+
+The following diagram illustrates the flow of logs from AP to the ELK stack:
+
+![AP with ELK](img/ug_ap_logging_elk.svg)
+
+### 5.1 Sending logs from AP to Elasticsearch with Logstash
+
+To set up the ELK stack, the first step is installing and configuring Logstash. This section details how to configure Logstash to receive logs from AP and send them to Elasticsearch for storage and analysis. If the ELK stack isn't installed, refer to the provided [Docker setup example](#54-example-docker-environment-of-harmony-ap-with-elk-stack) for a quick start.
+
+Logstash configuration consists of three main components:
+
+1. **Input**: Defines how Logstash receives log messages.
+2. **Filter**: Defines how Logstash parses and processes log messages.
+3. **Output**: Defines where Logstash sends processed messages.
+
+For additional details, refer to the [Logstash documentation](#Ref_LOGSTASH).
+
+#### 5.1.1 Input configuration
+
+The input configuration specifies how Logstash should receive log messages. Logstash supports various [input plugins](https://www.elastic.co/guide/en/logstash/current/input-plugins.html) that allow it to receive logs from different sources, we will focus on reading logs from log files and gelf as those are the most relevant for AP.
+
+##### 5.1.1.1 Reading logs from log files
+
+For systems that log to files, Logstash's [file input plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-file.html) can read logs directly from the filesystem. This plugin tails log files and supports log rotation.
+
+This method is recommended for the Linux package of Harmony AP
+
+Example configuration for reading logs from a file:
+
+```conf
+input {
+  file {
+    path => "/var/log/harmony-ap/domibus.log"
+    codec => json
+    start_position => "beginning"
+    type => "domibus"
+    sincedb_path => "/usr/share/logstash/sincedb"
+  }
+}
+```
+
+Here, we specify the log file's path, the codec for parsing log messages, the starting position when reading a file for the first time, the type of log messages for further processing, and the `sincedb` file path for tracking the log file's position. 
+
+##### 5.1.1.2 Receive logs over the network
+
+The [gelf input plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-gelf.html) enables Logstash to receive logs in GELF (Graylog Extended Log Format) format over the network. This method is commonly used for Docker containers and is recommended for the Harmony the AP Docker image.
+
+Example configuration for receiving GELF logs:
+
+```conf
+input {
+  gelf {
+    port => 12201
+  }
+}
+```
+
+This configuration listens for GELF messages on port `12201`, which is the default GELF port.
+
+To send logs to Logstash in GELF format from Docker, configure the Docker container's logging driver. For example:
+
+```yaml
+ harmony-ap:
+   image: niis/harmony-ap:2.4.0
```

Comment:
When it comes to the version of the Harmony Docker image, consider using a version neutral approach in the example snippets:

> image: niis/harmony-ap:\<image tag\>